### PR TITLE
Add VS Code Development Container settings

### PR DESCRIPTION
**Pull request by pellared**
## Why

An automated way to have a [dev environment inside a container](https://code.visualstudio.com/docs/remote/containers) that would not clutter your host OS.
It makes it also possible to test/build stuff on Windows as it was on Linux. Personally, I use Win10 with WSL2. The only stuff currently that I saw not working is IPv6. Here is why: https://docs.docker.com/config/daemon/ipv6/
This config may also be used in future to develop using [GitHub Codespaces](https://github.com/features/codespaces) in the browser once it will hit the production.

I feel that it is cool for an open-source repo to have the "environment setup" as easy as possible to welcome contributions. Especially bugfixes 😉 

## What

Add VS Code development container setting including Docker in Docker and .NET 5.0 with additional .NET Core runtimes.
The development container does not contain the Native (C++) stuff, however, Docker and Docker Compose are available inside. I can work in improving the Dockerfile to make writing C++ more pleasant, but I do not know if it is worth the effort.

The stuff required to this are:
- Docker
- VS Code with [Remote Development extensions](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack)

Little more info: https://github.com/microsoft/vscode-dev-containers

### CODE REVIEW

**Comment by pjanotti** [comment]
> work in improving the Dockerfile to make writing C++ more pleasant, but I do not know if it is worth the effort.

Let's hold off on that for the time being.

**Review by pjanotti — COMMENTED** [review]
Change itself LGTM, however, I would like to see the setup/steps in a new md file under `docs/`.

**Comment by pellared** [comment]
> Change itself LGTM, however, I would like to see the setup/steps in a new md file under `docs/`.

Agree. Done 😉

**Review by dszmigielski — APPROVED** [review]

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
 3. Submit a PR updating the version with the changes above.
 4. Approve and merge PR above.
-5. Create the label for the release and add all the CircleCI artifacts from the PR above.
-There are artifacts: under `Alpine`, `Linux`, and `Windows`.
-6. Publish the release on GH.
-7. Publish `SignalFx.NET.Tracing.Azure.Site.Extension.*.nupkg`, `SignalFx.Tracing.*.nupkg`, and `SignalFx.Tracing.OpenTracing.*.nupkg` (from `Windows` artifacts) to https://www.nuget.org/
-
-*Modified by SignalFx*
\ No newline at end of file
+5. Publish a new GitHub Release and describe what has changed. CircleCI will automatically add artifacts and publish a new NuGet package.
```
Comment (pjanotti) [inline docs/CONTRIBUTING.md:25]: Good catch :)

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
+
+## Development Container
+
+The repository contains configuration for [developing inside a Container](https://code.visualstudio.com/docs/remote/containers) using [Visual Studio Code Remote - Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers).
+
+- [Installation](https://code.visualstudio.com/docs/remote/containers#_installation) 
+
+The Development Container configuration mixes [Docker in Docker](https://github.com/microsoft/vscode-dev-containers/tree/master/containers/docker-in-docker) and [C# (.NET)](https://github.com/microsoft/vscode-dev-containers/tree/master/containers/dotnet) definitions. Use the [Official Development Container Definitions](https://github.com/microsoft/vscode-dev-containers) if more is needed.
+
+There may be a lot of errors, because some projects target .NET Framework. Switch to `Datadog.Trace.Minimal.sln` or `Datadog.Trace.Native.sln` using `F1 -> OmniSharp: Select Project` in Visual Studio Code to load a subset of projects which work without any issues. You can also try building the projects which have errors as it sometimes helps.
```
Comment (pjanotti) [inline docs/CONTRIBUTING.md:13]: I suppose one can also install the target packages to work around these issues, anyway, IIRC the error message itself proposes that as a solution.

**Review by pjanotti — APPROVED** [review]
LGTM.

File: docs/CONTRIBUTING.md (hunk 2 above)
Comment (pellared) [reply docs/CONTRIBUTING.md:13]: Could not fix all issues TBH. We can try if it would be ever needed 😉

**Pull request merged**